### DISCUSSION TITLE
Limit messageInputBar to bounds of modal presented card style VC on iPad

### DISCUSSION THREAD

**Discussion by Granluke** [MessageKit/MessageKit#1693] [Q&A] [answered]
How can I limit the width of the presented input bar to the view controller bounds?

On iPad when presenting a modal view controller as a card the input bar is the width of the fullscreen
![IMG_0D0265E68A97-1](https://user-images.githubusercontent.com/23178302/161837092-971eda1e-9db1-4407-a211-e59a1b6decc5.jpeg)
 
How can I prevent this? I already tried it with the `keyboardManager` and adding the input bar as a subview to my vc, but to no avail. Any help would be appreciated.

**Comment by martinpucik** [comment, marked as answer]
@Granluke you should be able to do this with v4 now
